services/packages/packages.go

```diff
@@ -657,3 +659,29 @@ func RemoveAllPackages(ctx context.Context, userID int64) (int, error) {
 	}
 	return count, nil
 }
+
+func LinkPackageToRepository(ctx context.Context, doer *user_model.User, p *packages_model.Package, repoID int64) error {
+	if repoID != 0 {
+		repo, err := repo_model.GetRepositoryByID(ctx, repoID)
```

Comment:
This MUST check if the repo belongs to the package owner. And there should be a test for linking a repo of another user.

Reply:
To avoid further back and forth on permissions, and me doing best-effort guesses, can you give me a detailed run down on who should be allowed to do what?